src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Testing/MsTestV2/TestMethodRunnerExecuteIntegration.cs

```diff
@@ -25,9 +25,14 @@ public class TestMethodRunnerExecuteIntegration
     {
         private const string IntegrationName = nameof(IntegrationIds.MsTestV2);
         private static readonly IntegrationInfo IntegrationId = IntegrationRegistry.GetIntegrationInfo(IntegrationName);
+        private static readonly Tracer TestTracer;
+        private static readonly string ServiceName;
 
         static TestMethodRunnerExecuteIntegration()
         {
+            TestTracer = Tracer.Instance;
```

Comment:
This assume that users never replace `Tracer.Instance` with a new tracer instance (possibly with a different service name). Is that something we should worry about?

Reply:
in a testing scenario I cannot think why we should allow to replace the tracer instance. I did this mostly to avoid our test suite messing with the tracer being used to report the tests.

Reply:
It's probably fine. I just get nervous about saving the tracer instance. We're been bitten by bugs like these before.

> I cannot think why we should allow to replace the tracer instance.

We can't really stop users from doing that, can we? 

> to avoid our test suite messing with the tracer being used to report the tests

Out of scope of this PR, but we should make sure that we never use `Tracer.Instance` for testing purposes (only use it for for tracing). Tests should create their own `Tracer` instances if needed.

Reply:
> It's probably fine. I just get nervous about saving the tracer instance. We're been bitten by bugs like these before.

Do you know more about these bugs? To make sure I'm not missing something. Also I can do something like:
```csharp
public Tracer TestTracer => _testTracer ?? Tracer.Instance;
```
To ensure the TestTracer is not null.

> 
> > I cannot think why we should allow to replace the tracer instance.
> 
> We can't really stop users from doing that, can we?

If the `TestTracer` is fixed, if the user changes the `Tracer.Instance` later doesn't affect the test integration. I meant I cannot think a scenario where the user would change the TestTracer instance on purpose.

> 
> > to avoid our test suite messing with the tracer being used to report the tests
> 
> Out of scope of this PR, but we should make sure that we never use `Tracer.Instance` for testing purposes (and only use it for tracing). Tests should create their own `Tracer` instances if needed.

I agree, we currently lack of this isolation, tests can replace the `Tracer.Instance` and also there are some tests that relies on static properties (running in parallel is a source of race conditions).

Reply:
>Do you know more about these bugs?

Users can set configuration in several ways: environment variables, `web.config` files, or in code. For the latter, users need to do something like this (normally during application startup):
```csharp
var settings = new TracerSettings
               {
                   // ... set some properties ...
               };

Tracer.Instance = new Tracer(settings);
```

We had a bug in the ASP.NET Core integration where we cached the value of `Tracer.Instance`, but users who were doing configuration in code would replace `Tracer.Instance` _after_ that, so we did not see any changes to the settings. Until we fixed the bug, the workaround was to always set configuration using environment variables.

Reply:
I think for now we should store the `TestTracer` instance and revisit this issue when we change the tests that uses the `Tracer.Instance` in another PR. Because not being able to change the default configuration of the TestTracer is not big issue right now.

Reply:
Sounds good to me. Resolving this thread.